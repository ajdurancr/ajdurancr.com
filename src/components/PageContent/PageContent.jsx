import React from 'react'
import propTypes from 'prop-types'

import styles from './PageContent.styles.sass'

const PageContent = ({ content }) => (
  <>
    {Boolean(content.name) && <h1 className={styles.title}>{content.name}</h1>}
  </>
)

PageContent.defaultProps = {
  content: {},
  site: {},
}

PageContent.propTypes = {
  content: propTypes.shape({
    sections: propTypes.arrayOf(propTypes.object),
    name: propTypes.string,
    url: propTypes.string,
  }),
  site: propTypes.shape({
    id: propTypes.number,
    name: propTypes.string,
  }),
}

export default PageContent
