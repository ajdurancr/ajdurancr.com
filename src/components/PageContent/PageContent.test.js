import React from 'react'
import { shallow } from 'enzyme'

import PageContent from './PageContent'

describe('PageContent', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<PageContent content={{ name: 'Test Page' }} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders correctly with default props', () => {
    const wrapper = shallow(<PageContent />)

    expect(wrapper).toMatchSnapshot()
  })
})
