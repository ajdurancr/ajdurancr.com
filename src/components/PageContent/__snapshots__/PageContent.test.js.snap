// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContent renders correctly 1`] = `
<Fragment>
  <h1
    className="title"
  >
    Test Page
  </h1>
</Fragment>
`;

exports[`PageContent renders correctly with default props 1`] = `<Fragment />`;
